src/datadog/remote_config/remote_config.cpp

```diff
@@ -173,28 +159,41 @@ std::vector<ConfigMetadata> RemoteConfigurationManager::process_response(
     if (client_configs_it == json.cend()) {
       if (!applied_config_.empty()) {
         std::for_each(applied_config_.cbegin(), applied_config_.cend(),
-                      [this, &config_updates](const auto it) {
-                        auto updated = revert_config(it.second);
-                        config_updates.insert(config_updates.end(),
-                                              updated.begin(), updated.end());
+                      [this](const auto& it) {
+                        for (const auto& listener :
+                             listeners_per_product_[it.second.product]) {
+                          listener->on_revert(it.second);
+                        }
                       });
         applied_config_.clear();
       }
-      return config_updates;
+
+      for (const auto& listener : listeners_) {
+        listener->on_post_process();
+      }
+      return;
```

Comment:
configurations should be reverted when they are not included in a response, not when **all** configurations are removed from a response. It is possible to simultaneously apply a new configuration and revert a previous configuration.

Reply:
This part of the code is executed when `client_configs` is absent from the response. I can find on the spec:
> client_configs: List of configurations path that the tracer MUST apply. 

My understand is configurations outside of `client_configs` are not intended for the tracer. Since the request is made with anything necessary to identify a tracer (`is_tracer` and `client_tracer`), I would expect the response to target the tracer. Am I missing something?

Reply:
This seems correct to me, I see that the code below handles the case @cataphract is describing...

Reply:
Ah yes, I see there are two paths that handle `on_revert`, and I hadn't read into the latter one later in the function. I would refactor this to merge the two branches together. Also, this function is on the long side, it could be refactored.